pkg/api/jobrunintervals/job_run_intervals.go

```diff
@@ -15,26 +15,46 @@ import (
 )
 
 // JobRunIntervals fetches intervals for a given job run.
-func JobRunIntervals(gcsClient *storage.Client, dbc *db.DB, jobRunID int64, logger *log.Entry) (*apitype.EventIntervalList, error) {
+// It fetches from the prow job's GCS bucket path contructed using the prow job name and jobID
+// using one of these methods:
+// 1) looking up the url given the jobRunID and extracting the prow job name (we need to wait until the sippyDB is populated)
+// 2) using a GCS path that was calculated and passed in (we can retrieve intervals immediately)
+// If the GCS path could not be calculated, it will be empty.
+func JobRunIntervals(gcsClient *storage.Client, dbc *db.DB, jobRunID int64, gcsPath string, logger *log.Entry) (*apitype.EventIntervalList, error) {
 
-	// Right now, we need the job run in our DB to fetch its URL, so we can find the GCS
-	// bucket. This means until sippy imports the job, you will not be able to fetch
-	// its intervals.
-	// However, long term we expect these to live in an external system, and the prow job ID
-	// should be all we need to look it up. At that point the limitation should be removed.
-	jobRun, _, err := api.FetchJobRun(dbc, jobRunID, logger)
-	if err != nil {
-		logger.WithError(err).Error("error querying job run")
-		return nil, err
+	var gcsJobRun *gcs.GCSJobRun
+	var intervalFiles [][]string
+	if len(gcsPath) > 0 {
+		log.WithField("gcsPath", gcsPath).Debug("calculated gcs path from job attributes")
+		bkt := gcsClient.Bucket(gcs.OpenshiftGCSBucket)
+		gcsJobRun = gcs.NewGCSJobRun(bkt, gcsPath)
+		intervalFiles = gcsJobRun.FindAllMatches([]*regexp.Regexp{gcs.GetIntervalFile()})
 	}
 
-	parts := strings.Split(jobRun.URL, gcs.OpenshiftGCSBucket)
-	path := parts[1][1:]
-	log.WithField("path", path).Debug("calculated gcs path")
-	bkt := gcsClient.Bucket(gcs.OpenshiftGCSBucket)
-	gcsJobRun := gcs.NewGCSJobRun(bkt, path)
+	// If there we couldn't calculate the gcsPath, then use the database if we can.
+	if len(gcsPath) == 0 ||
+		// If there is no match, there will still be one empty item.
+		// This means either the GCS bucket path is wrong or there are no interval files.
+		(len(intervalFiles) == 1 && len(intervalFiles[0]) == 0) {
```

Comment:
I don't think we should lump the "no gcsPath provided" case together with gcsPath provided but wrong and fall back to looking up from the db. If you gave us a path, we should try to use it and fail if it's wrong, not silently try to do something else.

Please refactor to an if len(gcsPath) > 0 and an else.